### DISCUSSION TITLE
Cask where download artefact is gzipped, but without .gz extension

### DISCUSSION THREAD

**Discussion by chrishowell** [Homebrew/discussions#1460] [Tap maintenance and Homebrew development]
I'm trying to write a Cask for a resource that is behind a CloudFlare CDN, it looks like they gzip the resource even if the requester doesn't specify that it can accept compression.
The resource is a .dmg file, but Cloudflare gzips it and doesn't give it the .gz extension, as such the default behaviour is for Homebrew to download the resource and try and unzip it (using the magic number) but then gunzip fails with "unknown suffix -- ignored".
I see (at least) two options, but I'm not sure I can do either of them with the current Cask DSL: 

1. pass '--compressed' to Curl
2. rename the file before it's uncompressed

Does anyone have any other ideas, or is there actually a way of doing either of these already?

Thanks!!

**Comment by SMillerDev** [comment]
Is it zipping the file or just the connection?

**Reply by chrishowell** [reply]
the file itself is zipped. It guess the artefact should have the extension `.dmg.gz` I was also surprised, because if you download the link in Chrome it automatically uncompresses it, and the Curl command to download the file as retrieved from Chrome Developer Console includes `--compressed`.
